js/packages/common/src/actions/auction.ts

```diff
@@ -117,13 +117,24 @@ export enum PriceFloorType {
 }
 export class PriceFloor {
   type: PriceFloorType;
-  // It's an array of 32 u8s, when minimum, only first 4 are used (a u64), when blinded price, the entire
+  // It's an array of 32 u8s, when minimum, only first 8 are used (a u64), when blinded price, the entire
   // thing is a hash and not actually a public key, and none is all zeroes
-  hash: PublicKey;
+  hash: Uint8Array;
 
-  constructor(args: { type: PriceFloorType; hash: PublicKey }) {
+  minPrice?: BN;
+
+  constructor(args: {
+    type: PriceFloorType;
+    hash?: Uint8Array;
+    minPrice?: BN;
+  }) {
     this.type = args.type;
-    this.hash = args.hash;
+    this.hash = args.hash || new Uint8Array(32);
+    if (this.type === PriceFloorType.Minimum) {
+      if (args.minPrice) {
+        this.hash.set(args.minPrice.toArrayLike(Buffer, 'le', 8), 0);
```

Comment:
So this sets the first 8 entries, and leaves the rest as 0s?

Reply:
correct.

Reply:
Love it. Let's merge this bad boy.